### PR TITLE
Replace Joda with Java 8 times

### PR DESCRIPTION
**Pull request by regiskuckaertz**
I thought there's no more good reasons to use Joda now?

screen shot 2017-10-02 at 17 40 24

### CODE REVIEW

File: src/main/scala/com.gu.contentapi.client/utils/CapiModelEnrichment.scala [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 object CapiModelEnrichment {
 
   implicit class RichCapiDateTime(val cdt: CapiDateTime) extends AnyVal {
-    def toJodaDateTime: DateTime = new DateTime(cdt.dateTime)
+    def toLocalDateTime: LocalDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(cdt.dateTime), ZoneOffset.UTC)
   }
 
-  implicit class RichJodaDateTime(val dt: DateTime) extends AnyVal {
-    def toCapiDateTime: CapiDateTime = CapiDateTime.apply(dt.getMillis, dt.toString(ISODateTimeFormat.dateTime()))
+  implicit class RichJodaDateTime(val dt: LocalDateTime) extends AnyVal {
```
Comment (mchv) [inline src/main/scala/com.gu.contentapi.client/utils/CapiModelEnrichment.scala]: `RichJodaDateTime` need to be renamed `RichLocalDateTime`

**Comment by mchv** [comment]
Removing `joda` dependency is a very good idea 👏 
I have checked that the date format are the same, so this should be safe to be merged.

We probably want to add some notes on the changelog on how to convert from a `joda `DateTime` instance to a java 8 one, so that consumers that update to content-api can migrate without pain (without initially removing the dependency in their codebase).

Regarding versioning we should keep `models` and  the `client` in sync, so we should not release a new version only with this change.

**Comment by mchv** [comment]
👍

**Comment by tomrf1** [comment]
💯

**Comment by regiskuckaertz** [comment]
@mchv do I just bump the models version then?

**Comment by regiskuckaertz** [comment]
ah, I see, there's a few Joda instances in the models repo too. On it 👍

**Comment by regiskuckaertz** [comment]
One question though because I'm not sure: In Joda, `DateTime` represents a point in time _with_ a time zone. I've been using `LocalDateTime` in this PR, which is _without_ a time zone, because I thought CAPI stored times in the UTC format. Do I have to use `ZonedDateTime`/`OffsetDateTime` instead?

File: src/main/scala/com.gu.contentapi.client/Parameters.scala [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package com.gu.contentapi.client
 
-import org.joda.time.ReadableInstant
```
Comment (regiskuckaertz) [inline src/main/scala/com.gu.contentapi.client/Parameters.scala]: `ReadableInstant` is both implemented by `DateTime` and `Instant` in Joda, but the former is for human-readable times while the latter is for machine-readable times. I've opted for the latter here. (In Java 8, [they did not make that mistake](http://blog.joda.org/2009/11/why-jsr-310-isn-joda-time_4941.html))

**Comment by regiskuckaertz** [comment]
I've switched to `OffsetDateTime` as discussed.

**Comment by mchv** [comment]
👍

**Pull request merged**